src/main/scala/it/pagopa/interop/tenantprocess/api/impl/TenantApiServiceImpl.scala

```diff
@@ -235,8 +235,10 @@ final case class TenantApiServiceImpl(
       tenant         <- existingTenant
         .fold(createTenant(seed, Nil, now, getTenantKind(Nil, seed.externalId).fromAPI))(Future.successful)
       tenantKind     <- getTenantKindLoadingCertifiedAttributes(tenant.attributes, tenant.externalId)
-      updatedTenant  <- updateSelfcareId(tenant, tenantKind)
-    } yield updatedTenant.toApi
+      _              <- updateSelfcareId(tenant, tenantKind)
+      _              <- tenantManagementService.addTenantMail(tenant.id, seed.digitalAddress.toDependency)
+      tenant         <- tenantManagementService.getTenantById(tenant.id)
```

Comment:
because of the eventual consistency this could return an older version of the tenant (without selfcare id and mail), or not find it at all in the worst case.

IMO the cleanest alternative is to return just the id and let the caller use (if needed) any information that it already sent for this creation. This requires a minor alignment of the BFF (in `getSessionToken` function).

May I ask you to apply this change?

Reply:
Can I change addTenantMail and DeleteTenantMail from 204 to 200 and return the whole Tenant instead.
What do you think?

Reply:
your proposal makes totally sense, however returning the id is closer to the implementation of the ongoing refactor. This way we can minimize di difference between the two versions